warehouse/query-core/src/main/java/org/apache/commons/jexl2/parser/ASTDelayedPredicate.java

```diff
@@ -44,10 +44,18 @@ public String toString() {
     }
     
     /**
+     * Wrap a node in an ASTDelayedPredicate, if not already delayed
+     *
      * @param node
-     * @return
+     *            a JexlNode
+     * @return an ASTDelayedPredicate, or the original node if this node is already delayed
      */
-    public static ASTDelayedPredicate create(JexlNode node) {
+    public static JexlNode create(JexlNode node) {
+        
+        // Do not delay if this subtree is already delayed
+        if (QueryPropertyMarker.findInstance(node).isType(ASTDelayedPredicate.class) || ASTDelayedPredicate.isSubTreeAlreadyDelayed(node)) {
```

Comment:
It kind of looks like this `QueryPropertyMarker.findInstance(node).isType(ASTDelayedPredicate.class)` might already be handled if you just did `if (isSubTreeAlreadyDelayed(node))`. Not super used to query code though. Is that right? Or could the first check be true when the other check is false?

Reply:
I'm leaning towards closing this PR and working these cases into https://github.com/NationalSecurityAgency/datawave/pull/1297